### DISCUSSION TITLE
Clangd in Combination with cpplint oder sonarlint

### DISCUSSION THREAD

**Discussion by msMatix** [LazyVim/LazyVim#1480] [Q&A] [answered]
Hello All,

I have the following problem:
`warning: multiple different client offset_encodings detected for buffer, this is not supported yet
`
I have also looked through various forums and seen the post here. However, I get the warning again when I add a second linter like `sonarlint or cpplint.`

My configuration for `cpplint` is e.g. as follows:
```
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      setup = {
        clangd = function(_, opts)
          opts.capabilities.offsetEncoding = { "utf-16" }
        end,
      },
    },
  },
  -- does not work - null-ls is also archivied
  {
    "jose-elias-alvarez/null-ls.nvim",
    opts = function(_, opts)
      local nls = require("null-ls")
      opts.sources = opts.sources or {}
      vim.list_extend(opts.sources, {
        nls.builtins.diagnostics.cpplint,
      })
    end,
    dependencies = {
      "mason.nvim",
      opts = function(_, opts)
        opts.ensure_installed = opts.ensure_installed or {}
        vim.list_extend(opts.ensure_installed, { "cpplint" })
      end,
    },
  },
}
```

Can anyone help me with this?
Thanks a lot!

VG
Matix

**Comment by dpetka2001** [comment]
I believe this must go under `servers.clangd` and not `setup` if I'm not mistaken. You can also take a look at [clangd extra](https://github.com/amaanq/LazyVim/blob/d0807d61b5dcd68f9140026d8e7e15c5b29012bf/lua/lazyvim/plugins/extras/lang/clangd.lua#L67-L69).

**Reply by msMatix** [reply]
Ah I just see that utf-16 is already used. That means the fix is no longer necessary. Unfortunately, this still does not work.

**Reply by dpetka2001** [reply, marked as answer]
With the following 
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        clangd = {
          capabilities = {
            offsetEncoding = { "utf-16" },
          },
        },
      },
    },
  },
  -- does not work - null-ls is also archivied
  {
    "jose-elias-alvarez/null-ls.nvim",
    opts = function(_, opts)
      local nls = require("null-ls")
      opts.sources = opts.sources or {}
      vim.list_extend(opts.sources, {
        nls.builtins.diagnostics.cpplint,
      })
    end,
    dependencies = {
      "mason.nvim",
      opts = function(_, opts)
        opts.ensure_installed = opts.ensure_installed or {}
        vim.list_extend(opts.ensure_installed, { "cpplint" })
      end,
    },
  },
}
```
I don't get any warning in a `.c` file. I don't use the `clangd` from Lazyvim's extras. I just installed `clangd` and `cpplint` with the aforementioned configuration.

**Reply by msMatix** [reply]
Hey thanks a lot!
Unfortunately, that doesn't work for me either. I have also removed clangd extras. What exactly do you install in your setup (plugin, ...)?
Then I would try that again.

**Reply by dpetka2001** [reply]
I didn't install anything. I tried the above code in a fresh Lazyvim installation with all settings and plugins default, and only installing `clangd` and `cpplint`. Nothing more.

**Reply by msMatix** [reply]
ok so no it works. I tried it again and just did a reinstall of cpplint and clangd. 
And just a hint. If i enable the extras for clangd. The warning message is back again. So there must something be wrong. I think I will do it by my own as you did. :)

Thank you for your support.
